opensoundscape/audiomoth.py

```diff
@@ -115,3 +114,59 @@ def parse_audiomoth_metadata_from_path(file_path):
             )
         else:
             return parse_audiomoth_metadata(metadata)
+
+
+def _parse_audiomoth_comment_dt(comment):
+    """parses start times as written in metadata Comment field of AudioMoths
+
+    examples of Comment Field date-times:
+    19:22:55 14/12/2020 (UTC-5)
+    10:00:00 15/05/2021 (UTC)
+
+    note that UTC-5 is not parseable by datetime, hence custom parsing
+    also note day-month-year format of date
+
+    Args:
+        comment: the full comment string from an audiomoth metadata Comment field
+    Returns:
+        localized datetime object in timezone specified by original metadata
+    """
+    # extract relevant portion of comment
+    dt_str = comment.split("Recorded at ")[1].split(" by ")[0]
+
+    # handle formats like "UTC-5" or "UTC+0130"
+    if "UTC-" in dt_str or "UTC+" in dt_str:
+        marker = "UTC-" if "UTC-" in dt_str else "UTC+"
+        dt_str_utc_offset = dt_str.split(marker)[1][:-1]
+        if len(dt_str_utc_offset) <= 2:
+            dt_str_tz_str = f"{marker}{int(dt_str_utc_offset):02n}00"
+        else:
+            dt_str_tz_str = f"{marker}{int(dt_str_utc_offset):04n}"
+
+        dt_str = f"{dt_str.split(marker)[0]}{dt_str_tz_str})"
+    else:  #
+        dt_str = dt_str.replace("(UTC)", "(UTC-0000)")
+    dt = datetime.datetime.strptime(
+        dt_str, "%H:%M:%S %d/%m/%Y (%Z%z)"
+    )  # .astimezone(final_tz)
+    return dt
+
+
+def _parse_audiomoth_battery_info(comment):
+    """attempt to parse battery info from metadata comment
+
+    examples:
+    ...battery state was 4.7V.
+    ...battery state was less than 2.5V
+    ...battery state was 3.5V and temperature....
+
+    Args:
+        comment: the full comment string from an audiomoth metadata Comment field
+    Returns:
+        float of voltage or string describing voltage, eg "less than 2.5V"
+    """
+    battery_str = comment.split("battery state was ")[1].split("V")[0] + "V"
+    if len(battery_str) == 4:
+        return float(battery_str[:-1])
+    else:
+        return battery_str
```

Comment:
Why does this sometimes return the string including the V for 1 case, but not including the V for the other case?
```
>> _parse_audiomoth_battery_info("battery state was more than 4.5V")
>> 'more than 4.5V'

>>_parse_audiomoth_battery_info("battery state was 4.5V")
>>'4.5'
```
Is that what we want?

Reply:
the idea was that it would return floats of the voltage when possible (as it originally did) for convenience of the metadata's use, but then return the descriptive string when its not just a number.

Reply:
IDK if this is the best approach